### PR TITLE
[2009] Aleksandra Tanasijenko

### PR DESCRIPTION
**Pull request by alekstana**
I finally did the first iteration and I think it works pretty well.
I managed to manipulate "stop and reset buttons" I think and added and event listener for "// Start/Stop Button", but I am not sure if is correct.
I will continue working on this Lab tomorrow!

### CODE REVIEW

**Comment by alekstana** [comment]
I tried to debug...i didn't succeed :(

File: javascript/chronometer.js [hunk 1]
```diff
@@ -4,33 +4,38 @@ class Chronometer {
     this.intervalId = 0;
 
   }
-  startClick(callback) {
-    setInterval(()=>{
-      this.currentTime++
-    },1000)
+  startClick() {
+    this.intervalId = setInterval(function(){
```
Comment (IH-WebDev-TA-Remote) [inline javascript/chronometer.js:8]: Interval function needs to be an arrow function in order for the this keyword to target to the correct element (the classChronometer)

File: javascript/index.js [hunk 2]
```diff
@@ -9,20 +9,23 @@ let minDec = document.getElementById('minDec');
...
 
 function printTime() {
-  // ... your code goes here
+   printMinutes()
+  // function printSeconds()
 }
 
 function printMinutes() {
-  // ... your code goes here
+  minDec.innerHTML = Number(minutes[0])
```
Comment (IH-WebDev-TA-Remote) [inline javascript/index.js:22]: minutes doesn't exist in this code, the idea here was to use the methods available inside the chronometer element (getSeconds, getMinutes and twoDigitsNumber to get the needed results to update the DOM.

File: javascript/index.js [hunk 3]
```diff
@@ -9,20 +9,23 @@ let minDec = document.getElementById('minDec');
...
+  minUni.innerHTML = Number(minutes[1])
 }
 
 function printSeconds() {
-  // ... your code goes here
+secUni.innerHTML = Number(seconds[0])
```
Comment (IH-WebDev-TA-Remote) [inline javascript/index.js:27]: Same as above minutes segment

File: javascript/index.js [hunk 4]
```diff
@@ -31,35 +34,66 @@ function printMilliseconds() {
 }
 
 function printSplit() {
-  // ... your code goes here
+  btnRight.addEventListener('click', () => {
```
Comment (IH-WebDev-TA-Remote) [inline javascript/index.js:37]: you are already assigning the addEventListener to the buttons in the last part of the code, these will wait for the button to be clicked and then run this function. So in here, the only code needed is the one that prints the result. No need to create a new addEventListener after you click the button.

File: javascript/index.js [hunk 5]
```diff
@@ -31,35 +34,66 @@ function printMilliseconds() {
...
+    if (btnRight.classList.contains('split')){
+    splits.innerHTML = splitValue
```
Comment (IH-WebDev-TA-Remote) [inline javascript/index.js:39]: Similar to the printMinutes and printSeconds comments, splitValue does not exist here, but you can use the chronometer splitClick method for this. you can call it simply doing: chronometer.splitClick()

File: javascript/index.js [hunk 6]
```diff
@@ -31,35 +34,66 @@ function printMilliseconds() {
...
+  btnRight.innerHTML = "RESET"
 }
 
 // Start/Stop Button
 btnLeft.addEventListener('click', () => {
-  // ... your code goes here
+  if (btnLeft.classList.contains('start')){
+  setStopBtn()
+  setSplitBtn()
+  chronometer.startClick()
```
Comment (IH-WebDev-TA-Remote) [inline javascript/index.js:73]: The concept of callback and it's implementation was really hard to understand and apply, but it is essential for the code to work. Basically you have a function that updates the DOM (printTime) and you want that to be executed every second, so you want it to run inside the interval. So by passing the function as a parameter here and ten invoke the parameter inside the interval you will be able to achieve that.

**Comment by IH-WebDev-TA-Remote** [comment]
Overall good effort, You finished all iteration 1 but it seems there were some issues preventing everything from updating in iteration 2, check all comments and also the solution we will send later. Also if something is not clear please let us know and we will explain it again :)

**Pull request closed without merging**